File: surface/widgets/swiftui/components/widgets/complex-inputs/RichTextEditor.swift
// ============================================================
// Clef Surface SwiftUI Widget — RichTextEditor
//
// Rich text editor with formatting toolbar. Supports bold,
// italic, and basic text formatting actions via a toolbar.
// ============================================================

import SwiftUI

// --------------- Component ---------------

/// RichTextEditor view with formatting toolbar.
///
/// - Parameters:
///   - value: Binding to the text content.
///   - placeholder: Placeholder text.
///   - enabled: Whether the editor is enabled.
struct RichTextEditorView: View {
    @Binding var value: String
    var placeholder: String = "Start typing..."
    var enabled: Bool = true

    var body: some View {
        VStack(spacing: 0) {
            // Formatting toolbar
            HStack(spacing: 4) {
                FormatButton(icon: "bold", label: "Bold")
                FormatButton(icon: "italic", label: "Italic")
                FormatButton(icon: "underline", label: "Underline")
                Divider().frame(height: 20)
                FormatButton(icon: "list.bullet", label: "Bullet list")
                FormatButton(icon: "list.number", label: "Numbered list")
                Divider().frame(height: 20)
                FormatButton(icon: "link", label: "Link")
                Spacer()
            }
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(Color(.systemGray6))

            Divider()

            // Text editor
            ZStack(alignment: .topLeading) {
                TextEditor(text: $value)
                    .disabled(!enabled)
                    .frame(minHeight: 120)

                if value.isEmpty && !placeholder.isEmpty {
                    Text(placeholder)
                        .foregroundColor(.secondary)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 10)
                        .allowsHitTesting(false)
                }
            }
        }
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(Color(.systemGray4), lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: 8))
    }
}

private struct FormatButton: View {
    let icon: String
    let label: String

    var body: some View {
        SwiftUI.Button(action: {}) {
            Image(systemName: icon)
                .font(.subheadline)
                .frame(width: 28, height: 28)
        }
        .buttonStyle(.plain)
        .accessibilityLabel(label)
    }
}

File: surface/widgets/nativescript/components/widgets/complex-inputs/RichTextEditor.ts
// ============================================================
// Clef Surface NativeScript Widget — RichTextEditor
//
// Rich text editing with formatting toolbar.
// ============================================================

import { StackLayout, Label, TextView, Button } from '@nativescript/core';

export interface RichTextEditorProps {
  value?: string;
  placeholder?: string;
  disabled?: boolean;
  readOnly?: boolean;
  toolbar?: ('bold' | 'italic' | 'underline' | 'heading' | 'list' | 'link')[];
  label?: string;
  onChange?: (value: string) => void;
}

export function createRichTextEditor(props: RichTextEditorProps): StackLayout {
  const {
    value = '', placeholder = 'Start writing...',
    disabled = false, readOnly = false,
    toolbar = ['bold', 'italic', 'underline'],
    label, onChange,
  } = props;

  const container = new StackLayout();
  container.className = 'clef-widget-rich-text-editor';

  if (label) {
    const lbl = new Label();
    lbl.text = label;
    container.addChild(lbl);
  }

  const toolbarRow = new StackLayout();
  toolbarRow.orientation = 'horizontal';
  toolbarRow.accessibilityRole = 'toolbar';
  for (const action of toolbar) {
    const btn = new Button();
    btn.text = action.charAt(0).toUpperCase();
    btn.className = 'clef-rte-toolbar-btn';
    btn.isEnabled = !disabled && !readOnly;
    btn.accessibilityLabel = action;
    toolbarRow.addChild(btn);
  }
  container.addChild(toolbarRow);

  const editor = new TextView();
  editor.text = value;
  editor.hint = placeholder;
  editor.isEnabled = !disabled;
  editor.editable = !readOnly;
  editor.height = 200;
  editor.accessibilityLabel = label || 'Rich text editor';
  editor.on('textChange', (args) => onChange?.(args.object.text));
  container.addChild(editor);

  return container;
}

export default createRichTextEditor;
